src/slices/GiveThunk.ts

```diff
@@ -75,7 +76,11 @@ export const changeApproval = createAsyncThunk(
     }
 
     const signer = provider.getSigner();
-    const sohmContract = new ethers.Contract(addresses[networkID].SOHM_V2 as string, ierc20Abi, signer);
+    const sohmContract = new ethers.Contract(
+      SOHM_ADDRESSES[networkID as keyof typeof SOHM_ADDRESSES] as string,
```

Comment:
Why not have a helper function for this?
```
getSOhmAddress(networkId): string => {
}
```

It can handle the casting of `networkID` and cast the return value to a string too.

Reply:
I'd argue no. Typecasting isn't a good thing to do in TS, especially if we are obscuring its use inside a helper function. The TS warning is often there for a reason, see previous conversation for explanation why.